src/main/java/wolox/training/repositories/BookRepository.java

```diff
@@ -14,4 +17,9 @@
 
 	Optional<Book> findFirstByIsbn(@NotBlank String isbn);
 
+//	List<Book> findByPublisherAndGenreAndYear(@NotBlank String publisher, @NotBlank String genre,
+//    @NotBlank String year);
+	@Query("select u from Book u where (u.publisher = :publisher or null = :publisher) and (u.genre = :genre or null = :genre) and (u.year = :year or null = :year)")
```

Comment:
Los = null deberían ir con **"is null"** porque con = null puede haber motores de bases de datos que no lo interpreten y devuelvan resultados inesperados.